projects/opendr_ws/src/perception/scripts/audiovisual_emotion_recognition.py

```diff
@@ -35,22 +36,22 @@
 class AudiovisualEmotionNode:
 
     def __init__(self, input_video_topic="/usb_cam/image_raw", input_audio_topic="/audio/audio",
-                 annotations_topic="/opendr/audiovisual_emotion", buffer_size=3.6, device="cuda"):
+                 output_emotions_topic="/opendr/audiovisual_emotion", buffer_size=3.6, device="cuda"):
         """
         Creates a ROS Node for audiovisual emotion recognition
         :param input_video_topic: Topic from which we are reading the input video. Expects detected face of size 224x224
         :type input_video_topic: str
         :param input_audio_topic: Topic from which we are reading the input audio
         :type input_audio_topic: str
-        :param annotations_topic: Topic to which we are publishing the predicted class
-        :type annotations_topic: str
+        :param output_emotions_topic: Topic to which we are publishing the predicted class
+        :type output_emotions_topic: str
         :param buffer_size: length of audio and video in sec
         :type buffer_size: float
         :param device: device on which we are running inference ('cpu' or 'cuda')
         :type device: str
         """
 
-        self.publisher = rospy.Publisher(annotations_topic, Classification2D, queue_size=10)
+        self.publisher = rospy.Publisher(output_emotions_topic, Classification2D, queue_size=10)
 
         video_sub = message_filters.Subscriber(input_video_topic, ROS_Image)
         audio_sub = message_filters.Subscriber(input_audio_topic, AudioData)
```

Comment:
Hi, I have tried your changes but the rosnode still creates the error for me. I think when you start the rosnode for camera and audio first, then even if you move the subscriber initialization to the `listen` node, there are still some delays between the audiovisual node initialization and calling the `listen` method. This will cause the error.
Other nodes I fixed also have this problem. I avoid the problem by initializing the perception node first and then the input nodes. In this way, the publisher will not publish any message before the `listen` method is called.

Reply:
It seems weird, i tried again and the changes seem to fix the error on my end. Here's what i did exactly:

Constructor:
```python
    def __init__(self, input_video_topic="/usb_cam/image_raw", input_audio_topic="/audio/audio",
                 output_emotions_topic="/opendr/audiovisual_emotion", buffer_size=3.6, device="cuda"):
        ...
        self.publisher = rospy.Publisher(output_emotions_topic, Classification2D, queue_size=10)

        self.input_video_topic = input_video_topic  # Changed line
        self.input_audio_topic = input_audio_topic  # Changed line

        self.bridge = ROSBridge()

        # Initialize the gesture recognition
        self.avlearner = AudiovisualEmotionLearner(device=device, fusion='ia', mod_drop='zerodrop')
        ...
```

listen:
```python
    def listen(self):
        ...
        rospy.init_node('opendr_audiovisualemotion_recognition', anonymous=True)

        video_sub = message_filters.Subscriber(self.input_video_topic, ROS_Image)
        audio_sub = message_filters.Subscriber(self.input_audio_topic, AudioData)
        # synchronize video and audio data topics
        ts = message_filters.ApproximateTimeSynchronizer([video_sub, audio_sub], 10, 0.1, allow_headerless=True)
        ts.registerCallback(self.callback)

        rospy.loginfo("Audiovisual emotion recognition node started!")
        rospy.spin()
```

Could you try this and maybe include a print message in listen to verify that indeed the node runs with the changes? I had similar issues where i needed to rebuild everything or even delete old build/install folders and rebuilding before my changes are indeed applied.

Reply:
Hi, I misunderstood your idea. I have used your suggestion and updated the code. Now there is no error. But one thing I want to note here, other ROS 1 nodes I implemented that had been merged to `develop` branch might have the same errors. Do I need to open a new issue and fix them?

Reply:
Good news, thanks! To be honest i didn't encounter the error when i ran the other nodes, but i was expecting it based on previous experience. If you are certain that the error is indeed present, i think you can fix it and open a new PR with the fixes straight away.